### DISCUSSION TITLE
Set SumartraPDF for previews after Adobe steals it after each update

### DISCUSSION THREAD

**Discussion by brinkdinges** [sumatrapdfreader/sumatrapdf#3584] [Ideas]
I love how fast the PDF preview in Windows Explorer is when I set Sumatra as the reader to use. But it seems like Adobe Reader updates every few weeks and overwrites that setting again.

Right now I re-download and re-install Sumatra every time this happens, without removing the previous (and identical) version to fix the preview. Would it be possible to add a button in the settings to fix the preview setting in Explorer?

**Comment by dom6770** [comment]
Yes, this is annoying. I wish Adobe would add an option to disable the preview handler, but welp.

The preview handler is controlled by a registry item.
You can get the current setting with PowerShell by executing:

`Get-ItemProperty -Path 'Registry::HKEY_CLASSES_ROOT\.pdf\shellex\{8895b1c6-b41f-4c1c-a562-0d564250836f}\' -Name "(default)"`

If it's set to "`{3D3B1846-CC43-42AE-BFF9-D914083C2BA3}`" SumatraPDF is the preview handler, the ID for Acrobat is "`{DC6EFB56-9CFA-464D-8880-44885D7DC193}`" You can get a list of handlers in `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\PreviewHandlers`

Now, you can either manually set the handler to SumatraPDF by executing

`Set-ItemProperty -Path 'Registry::HKEY_CLASSES_ROOT\.pdf\shellex\{8895b1c6-b41f-4c1c-a562-0d564250836f}\' -Name "(default)" -Value "{3D3B1846-CC43-42AE-BFF9-D914083C2BA3}"`

or create a Scheduled Job, which runs every 5 minute (or every :5 min):

```
$T = New-JobTrigger `
  -Once `
  -At (Get-Date).AddSeconds(60-(Get-Date).Second).AddMinutes(4-(Get-Date).Minute % 5 ) `
  -RepetitionInterval (New-TimeSpan -Minutes 5) `
  -RepeatIndefinitely

$O = New-ScheduledJobOption `
  -StartIfOnBattery `
  -ContinueIfGoingOnBattery

Register-ScheduledJob `
  -Name "SumatraPDF Preview Handler" `
  -Trigger $T `
  -ScheduledJobOption  $O `
  -ScriptBlock {Set-ItemProperty -Path 'Registry::HKEY_CLASSES_ROOT\.pdf\shellex\{8895b1c6-b41f-4c1c-a562-0d564250836f}\' -Name "(default)" -Value "{3D3B1846-CC43-42AE-BFF9-D914083C2BA3}" } `
  -RunNow
```

**Reply by GitHubRulesOK** [reply]
Since I am running both I let Acrobat preview as default file handler, it used to be Edge would steal the Allocation but Adobe upped their game to defeat that ! every 5 mins seems excessive if its just after Adobe updates. since its hooked into explorer then on login/startup with a manual override would seem adequate.

Previewers generally need the app to install so when you switch on off in adobe it reinstalls from its backup
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/3d2377ea-ccf2-4b0b-93ca-d70ebfd65658)


the problem is its a variable feast so my previewer is not showing under HKCR\\.pdf\shellex that is blank and just default.

so default MS Edge PDF previewer is listed as 3A84F9C2-6164-485C-A7D9-4B27F8AC009E
somewhere there was a utility to roll between pdf defaults edge/adobe/sumatrapdf but that may not have included previewer ?

**Comment by kjk** [comment]
I've opened https://github.com/sumatrapdfreader/sumatrapdf/issues/3879 to track this
